File: Submissions/#0365_div2/A/main.java
import java.util.*;

public class Main {

  public static void main (String[] args) {

    Scanner sc = new Scanner(System.in);
    int n = sc.nextInt();
    int m = 0;
    int c = 0;
    for (int i = 0; i < n; i++) {
      int a = sc.nextInt();
      int b = sc.nextInt();
      if (a != b) {
        if (a < b) {
          c++;
        } else {
          m++;
        }
      }
    }
    sc.close();

    if (m == c) {
      System.out.println("Friendship is magic!^^");
    } else if (c < m) {
      System.out.println("Mishka"); 
    } else {
      System.out.println("Chris");
    }

  }

}

File: Submissions/#0365_div2/A/main.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {

  int n;
  cin >> n;

  int m = 0;
  int c = 0;
  for (int i = 0; i < n; i++) {
    int a, b;
    cin >> a >> b;
    if (a != b) {
      if (a < b) {
        c++;
      } else {
        m++;
      }
    }
  }

  if (m == c) {
    cout << "Friendship is magic!^^" << endl;
  } else if (c < m) {
    cout << "Mishka" << endl; 
  } else {
    cout << "Chris" << endl;
  }

}